File: Python/8py 3/str_count.py
"""
|--------------------------------------------------------------------------
| All Star Code Challenge #18
|--------------------------------------------------------------------------
|
"""

def str_count(strng, letter):
   return len(list(filter(lambda i: i == letter , strng)))


str_count = str.count


def strCount(string, letter):
   return string.count(letter)


def str_count(strng, letter):
   return len(strng.split(letter)) - 1


def str_count(strng, letter):
   return sum([ch == letter for ch in strng])


print(str_count('hello', 'l'), 2)
print(str_count('hello', 'e'), 1)
print(str_count('codewars', 'c'), 1)
print(str_count('ggggg', 'g'), 5)
print(str_count('hello world', 'o'), 2)

File: JavaScript/8js 3/strCount.js
/*
|--------------------------------------------------------------------------
| All Star Code Challenge #18
|--------------------------------------------------------------------------
|
*/

function strCount(str, letter) {
   return str.split('').filter(i => i == letter).length;
}


const strCount = (w, l) => [...w].reduce((a, b) => a + +(b == l), 0);


function strCount(str, letter) {
   return str.split(letter).length - 1
}


function strCount(str, letter) {
   return [...str].filter(c => c === letter).length
}


function strCount(str, letter) {
   return str.length - str.replace(new RegExp(letter, "gi"), '').length;
}


console.log(strCount('Hello', 'o'), 1);
console.log(strCount('Hello', 'l'), 2);
console.log(strCount('', 'z'), 0);